doc/design/blob-uuid-1.md

```diff
@@ -0,0 +1,44 @@
+A blob is referenced by a key generation function. the value of this key is used to index the blob in the database table of identities and on the s3 storage. `s3://dandiarchive/blob/<3digitskey>/<next3digitskey>/<restofkey>`
+
+** database table of identities **: indexed by the key and contains columns for as many types of checksums as we want. these checksums constitute identities of the blob and should use functions that will always return the same value.
```

Comment:
@dchiquito - shouldn't this be indexed by the UUID with the ETag being one of the identities? only because two ETag's could collide (in theory, the probably for that is higher than the probability for UUID4s to collide on the same system)

Reply:
if we are to have dedicated columns for checksums, I thought they all could just go into "blobs" table with an index, UUID (unless UUID to be used as index, but I thought it is inferior), ETag (which we might want to name 'dandietag' or alike to make more explicit), sha256 columns to start with, i.e. that there would be no need in another table with 1-to-1 relationship to blobs table.  or what am I missing?

Reply:
i'm fine with `index, uuid, dandietag, sha256, blake3, <future checksums>` whether the index is built in or uuid, i don't care a whole lot. but there may be performance penalties for search, which would likely be given a checksum, find uuid, or given a uuid, find checksums.